From a6795e9ee3b642adc77a794455c11b95114a767b Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Tue, 14 Nov 2023 14:07:59 +0800
Subject: [PATCH] locallogin: allow /bin/login to read /etc/shadow

Update policy to fix login error:
type=AVC msg=audit(1699941637.635:76): avc:  denied  { read } for
pid=435 comm="login" name="shadow" dev="sda2" ino=533
scontext=system_u:system_r:local_login_t:s0-s15:c0.c1023
tcontext=system_u:object_r:shadow_t:s0 tclass=file permissive=1

Upstream-Status: Inappropriate [wrlinux specific]

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/system/locallogin.te | 1 +
 1 file changed, 1 insertion(+)

diff --git a/policy/modules/system/locallogin.te b/policy/modules/system/locallogin.te
index 7728de804..108f6268f 100644
--- a/policy/modules/system/locallogin.te
+++ b/policy/modules/system/locallogin.te
@@ -128,6 +128,7 @@ auth_manage_pam_runtime_dirs(local_login_t)
 auth_manage_pam_runtime_files(local_login_t)
 auth_manage_pam_console_data(local_login_t)
 auth_domtrans_pam_console(local_login_t)
+auth_read_shadow(local_login_t)
 
 init_dontaudit_use_fds(local_login_t)
 
-- 
2.25.1

